minizip-1.1/zip.c

```diff
@@ -977,8 +981,9 @@ int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_ex
 	  err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC1, 4);
 	  if (err==ZIP_OK)
 	  {
-	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC2, 4);
-	  }
+        u_int32_t tiapiheader = tiapimajor << 24 | tiapiminor << 16 | 0x504C;
```

Comment:
This is wrong - tiapimajor and tiapiminor are two bytes each.

I would like to have it converted to hex and then written.

So it should turn `tiapimajor = 0x05; tiapiminor = 0x00;` into `4c 50 30 35 30 30 LP0500`. The last two bytes are currently written as `LOCALHEADERMAGIC3`, which can be removed/replaced.